modules/gapi/test/common/gapi_video_tests_inl.hpp

```diff
@@ -88,6 +90,107 @@ TEST_P(BuildPyr_CalcOptFlow_PipelineTest, AccuracyTest)
     compareOutputsOptFlow(outOCV, outGAPI);
 }
 
+#ifdef HAVE_OPENCV_VIDEO
+TEST_P(BackgroundSubtractorTest, AccuracyTest)
+{
+    initTestDataPath();
+
+    cv::gapi::video::BackgroundSubtractorType opType;
+    double thr = -1;
+    std::tie(opType, thr) = typeAndThreshold;
+
+    cv::gapi::video::BackgroundSubtractorParams bsp(opType, histLength, thr,
+                                                    detectShadows, learningRate);
+
+    // G-API graph declaration
+    cv::GMat in;
+    cv::GMat out = cv::gapi::BackgroundSubtractor(in, bsp);
+    // Preserving 'in' in output to have possibility to compare with OpenCV reference
+    cv::GComputation c(cv::GIn(in), cv::GOut(cv::gapi::copy(in), out));
+
+    // G-API compilation of graph for streaming mode
+    auto gapiBackSub = c.compileStreaming(getCompileArgs());
+
+    // Testing G-API Background Substractor in streaming mode
+    try
+    {
+        gapiBackSub.setSource(gapi::wip::make_src<cv::gapi::wip::GCaptureSource>
+                                                      (findDataFile(filePath)));
+    }
+    catch (...)
+    { throw SkipTestException("Video file can't be opened."); }
+
+    cv::Ptr<cv::BackgroundSubtractor> pOCVBackSub;
+
+    if (opType == cv::gapi::video::TYPE_BS_MOG2)
+        pOCVBackSub = cv::createBackgroundSubtractorMOG2(histLength, thr,
+                                                         detectShadows);
+    else if (opType == cv::gapi::video::TYPE_BS_KNN)
+        pOCVBackSub = cv::createBackgroundSubtractorKNN(histLength, thr,
+                                                        detectShadows);
+
+    // Allowing 1% difference of all pixels between G-API and reference OpenCV results
+    testBackgroundSubtractorStreaming(gapiBackSub, pOCVBackSub, 1, 1, learningRate, testNumFrames);
+}
+
+TEST_P(KalmanFilterTest, AccuracyTest)
+{
+    GAPI_Assert(depth == CV_32F || depth == CV_64F);
+    GAPI_Assert(dDim > 0 && mDim > 0 && cDim >= 0);
+
+    cv::gapi::video::KalmanParams kp(dDim, mDim, cDim, depth);
+
+    //measurements vector
+    cv::Mat measure_vec(mDim, 1, depth);
+
+    //control vector
+    cv::Mat ctrl_vec;
+
+    if (cDim > 0)
+        ctrl_vec = Mat::zeros(cDim, 1, depth);
+    else if (cDim == 0)
+        ctrl_vec = Mat::zeros(dDim, 1, depth);
+
+    cv::randu(measure_vec, Scalar::all(-1), Scalar::all(1));
+
+    if (cDim > 0)
+        cv::randu(ctrl_vec, Scalar::all(-1), Scalar::all(1));
+
+    cv::Mat gapiKState(dDim, 1, depth);
+    cv::Mat ocvKState(dDim, 1, depth);
+
+    // G-API code
+    cv::GMat m, ctrl;
+    cv::GOpaque<bool> have_m;
+    cv::GMat out = cv::gapi::KalmanFilter(m, have_m, ctrl, kp);
+    cv::GComputation comp(cv::GIn(m, have_m, ctrl), cv::GOut(out));
+
+    cv::Mat gapiKalmanOut;
```

Comment:
Please, declare all the Mats before `G-API code` to make it clearer

Reply:
Ok. Changed.